ouroboros-consensus-cardano/tools/db-analyser/Main.hs

```diff
@@ -171,6 +187,7 @@ getCmdLine = execParser opts
 -------------------------------------------------------------------------------}
 
 analyse ::
```

Comment:
We may need to be clearer about what it means for the tool to "start from the given snapshot".

Specifically, in the current PR, if the command is to open the ChainDB and not just the ImmDB, then the ChainDB will initialize its LgrDB in the same way it always does, via the internal call to `initLedgerDB`.

https://github.com/input-output-hk/ouroboros-network/blob/d6826269569f239f262d8e72f2fd6f280334e4f7/ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs#L180-L237

Thus, if "the given snapshot" is not the one that `initLedgerDB` will choose, then there is room for the user to be surprised/confused.

Specifically, if the given snapshot is not the latest snapshot, then ChainDB will initialize from a later snapshot than "the given snapshot".

Since we're considering using this tool for benchmarking, for example, we need to be really clear about what phases it goes through. Moreover, it's possible the user will want more control that this PR currently offers. Notably, the user may want to specify which ledger snapshot the ChainDB's LedgerDB is initialized from. (At the moment, they'd have to do that by removing any later snapshots from the snapshot directory.)

-----

In fact, re-reading my comment above, I'm thinking what we probably want is a mode in which this tool opens the ImmDB and the VolDB but never opens the LgrDB. That'd be the easiest way to ensure that the "startup" phase of the tool involves no replay whatsoever, which I anticipate would be desirable for benchmarking uses.

Reply:
In particular the current `--help` of `"Initialize from stored ledger state stored at specific slot number"` is likely misleading for our current envisioned users/uses. I'd suggest instead

>Begin the analysis phase from the ledger state stored at the given slot number

Reply:
One short term option: we only allow `--initialize-from` if `--onlyImmutableDB` is also present. That avoids the potential for confusion without complicating this PR.

Reply:
Discussion I had with @nfrisby on this particular topic is below:

## Example

Imagine a scenario in which
your local blockchain has 1000
k  paramter is  200
immutable has 800 blocks, and there are 200 blocks which are still in volatile

in the db/ledger there are two snapshots:
* 700
* 500_db-analyser

User starts db-analyser with following parameters: --initialize-from 500 --num-blocks-to-process 400


What will happen:

1. ChainDB will find the latest snapshot (that is 700)
2. it will then start replying blocks from that snapshot up until tip (block 1000), in other words from 701 - 1000
3. That means that 300 blocks were processed while replay happened
4. After that we have:
* 200 blocks (volatile DB)
* we have the tip and its ExtLedgerState in memory
* we have intermediate ExtLedgerState in memory as well (for all 200 blocks)
5. This concludes ChainDB initialization


And now our db-analyser starts doing its thing:
1. it loads snapshot 500 (which holds both the ExtLedgerState and the slot no where it was taken)
2. it asks ChainDB to give it 400 blocks (300 from immutable and 100 from volatile - both loaded from disk)
3. and then for some commands (like --store-ledger) it will start applying those blocks to the ledger state - the one that was selected by --initialize-from (that will be 500)

## Proposed changes:

0. rename --onlyImmutableDB to --only-immutable-db
1. rename --intinialize-from to --analyse-from
2. make --analyse-from only an option if user calls the tool with --only-immutable-db

Reply:
I think these are all done. Approving shortly!